### PR TITLE
Fix 4.0

### LINKED ISSUE #548: Unable to install sonata/ecommerse on Symfony 4.1
Unable to install sonata/ecommerse on Symfony 4.1

composer require sonata-project/ecommerce:dev-master        
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Restricting packages listed in "symfony/symfony" to "4.1.*"
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - egeloen/ckeditor-bundle 6.0.x-dev requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 6.0.1 requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 6.0.0 requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 5.0.3 requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 5.0.2 requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 5.0.1 requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 5.0.0 requires symfony/dependency-injection ^2.7|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.6 requires symfony/dependency-injection ^2.2|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.5 requires symfony/dependency-injection ^2.2|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.4 requires symfony/dependency-injection ^2.2|^3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.3 requires symfony/dependency-injection ~2.2|~3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.2 requires symfony/dependency-injection ~2.2|~3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.1 requires symfony/dependency-injection ~2.2|~3.0 -> no matching package found.
    - egeloen/ckeditor-bundle 4.0.0 requires symfony/dependency-injection ~2.2|~3.0 -> no matching package found.
    - sonata-project/ecommerce dev-master requires sonata-project/formatter-bundle ^3.4 -> satisfiable by sonata-project/formatter-bundle[3.4.0, 3.4.1, 3.5.0, 3.x-dev].
    - sonata-project/formatter-bundle 3.4.0 requires egeloen/ckeditor-bundle ^4.0 || ^5.0 -> satisfiable by egeloen/ckeditor-bundle[4.0.0, 4.0.1, 4.0.2, 4.0.3, 4.0.4, 4.0.5, 4.0.6, 5.0.0, 5.0.1, 5.0.2, 5.0.3].
    - sonata-project/formatter-bundle 3.4.1 requires egeloen/ckeditor-bundle ^4.0 || ^5.0 -> satisfiable by egeloen/ckeditor-bundle[4.0.0, 4.0.1, 4.0.2, 4.0.3, 4.0.4, 4.0.5, 4.0.6, 5.0.0, 5.0.1, 5.0.2, 5.0.3].
    - sonata-project/formatter-bundle 3.5.0 requires egeloen/ckeditor-bundle ^4.0 || ^5.0 || ^6.0 -> satisfiable by egeloen/ckeditor-bundle[4.0.0, 4.0.1, 4.0.2, 4.0.3, 4.0.4, 4.0.5, 4.0.6, 5.0.0, 5.0.1, 5.0.2, 5.0.3, 6.0.0, 6.0.1, 6.0.x-dev].
    - sonata-project/formatter-bundle 3.x-dev requires egeloen/ckeditor-bundle ^4.0 || ^5.0 || ^6.0 -> satisfiable by egeloen/ckeditor-bundle[4.0.0, 4.0.1, 4.0.2, 4.0.3, 4.0.4, 4.0.5, 4.0.6, 5.0.0, 5.0.1, 5.0.2, 5.0.3, 6.0.0, 6.0.1, 6.0.x-dev].
    - Installation request for sonata-project/ecommerce dev-master -> satisfiable by sonata-project/ecommerce[dev-master].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.

Installation failed, reverting ./composer.json to its original content.

### PR DESCRIPTION
**Pull request by quisse**
I am targeting this branch, because not backwards compatible.

Closes #548 #531 #470 #511 

### Changed
- Allow SF4 formatter

## To do

- [ ] Update the tests
- [x] Update the documentation
- [ ] Add an upgrade note

## Subject

Allow SF4 formatter so this bundle works with SF4

### CODE REVIEW

**Comment by kunicmarko20** [comment]
Thank you for working on this!

**Comment by quisse** [comment]
Yes, will do. Shifted to another project for a while but I’ll get into it anytime soon.

File: docs/reference/installation.rst [hunk 1]
```diff
@@ -237,18 +230,32 @@ Then add the following bundles in your `kernel::registerBundles()` method (after
...
+    $ php bin/console doctrine:schema:update --force
 
 Create missing contexts:
 
 .. code-block:: bash
 
-    $ app/console sonata:classification:fix-context
-    $ app/console sonata:media:fix-media-context
+    $ php bin/console sonata:classification:fix-context
+    $ php app/console sonata:media:fix-media-context
```
Comment (pierredup) [inline docs/reference/installation.rst]: This needs to be the same as the line above

```suggestion
    $ php bin/console sonata:media:fix-media-context
```

File: docs/reference/bundles/price.rst [hunk 2]
```diff
@@ -18,23 +18,19 @@ Price bundle handles everything related to prices, (right now, only currencies,
...
-    class AppKernel extends Kernel
+    public function registerBundles()
     {
-        public function init()
-        {
-            bcscale(3); // or any other value greater than 0
-
-            ...
-
+        bcscale(3);
```
Comment (pierredup) [inline docs/reference/bundles/price.rst]: I think this function call would be better in the `boot` method of the kernel

```php
public function boot()
{
    bcscale(3); // or any other value greater than 0
    parent::boot();
}
```

File: docs/reference/installation.rst (hunk 1 above)
Comment (quisse) [reply docs/reference/installation.rst]: Must've overlooked that one. Thanks!

File: docs/reference/bundles/price.rst (hunk 2 above)
Comment (quisse) [reply docs/reference/bundles/price.rst]: Yes seems better to me too, thanks!

**Comment by greg0ire** [comment]
23 commits… please consider extracting commits that you know will pass the build into smaller PRs :pray:

**Comment by core23** [comment]
Can you please rebase @quisse

**Comment by greg0ire** [comment]
In case you were not convinced already, that high number of rebases and conflicts is one of the reasons why it would be better for both you, the contributor, and us, the maintainers, if you made smaller, easier to merge PRs. :point_up:

**Comment by quisse** [comment]
Yes, already got the point 🙂 will look into it. Got to admit, this pr started small and didn’t expect it would get this big. Just wanted to use this bundle with sf4

**Pull request closed without merging**